metricflow-semantics/metricflow_semantics/specs/instance_spec.py

```diff
@@ -46,6 +48,11 @@ def merge(*specs: Sequence[InstanceSpec]) -> List[InstanceSpec]:
     @property
     def qualified_name(self) -> str:
         """Return the qualified name of this spec. e.g. "user_id__country"."""
+        return self.structured_name.qualified_name
+
+    @property
+    def structured_name(self) -> StructuredLinkableSpecName:
```

Comment:
It seems like `StructuredLinkableSpecName` was only intended to support `LinkableSpecs`, not all `InstanceSpecs`. Do we need it for all `InstanceSpecs`? If so, we should rename it. If not, we should move this to the `LinkableSpec` class instead.